client/src/views/Create.vue

```diff
@@ -176,16 +198,28 @@ export default {
       this.forms = forms;
       this.$forceUpdate();
     },
-    updateModelMetaData(updatedMetaData) {
-      this.$emit('update-metadata', updatedMetaData);
+    updateModelName(name) {
+      this.model_metadata.model_name = name;
     },
     setCurrentlyVisible(name, id, prefix = '/create') {
       this.currentlyVisible = name;
       window.history.replaceState({}, '', `${prefix}#${id}`);
     },
-  },
-  created() {
-    this.createForms();
+    async handleSave() {
+      this.loading = true;
+      console.log(this.params);
+      if (this.$route.name === 'Edit') {
+        await this.$store.updateModel(this.$route.params.id, this.params);
+        if (this.model_metadata.model_name !== this.$route.params.id) {
+          await this.$store.cloneModel(this.$route.params.id, this.model_metadata.model_name);
+          await this.$store.deleteModel(this.$route.params.id);
+        }
```

Comment:
Letting people modify the name of their model from the 'forms' page seems a little weird to me. Maybe, like fig_type, model_name should only be editable from top level (list view of models) and forms present here could just be related to the model itself instead of the metadata?

Reply:
Also rename functionality should probably be the domain of $store instead of here. Potentially reusable logic.

Reply:
@Tengelma ya, rename still needs to be added into `Store`.

Reply:
The location can be changed later. It's not part of the scope of this change.